src/map/utils/diggingutils.cpp

```diff
@@ -0,0 +1,490 @@
+	}
+
+/************************************************************************
+*																		*
+*	Add to the character's Chocobo Dig count							*
+*																		*
+************************************************************************/
+
+	void UpdateZoneItemsDug(CCharEntity* PChar)
+	{
+		int8 value = 1;
+		int8 ZoneDigCount;
+		ZoneDigCount = GetZoneItemsDug(PChar);
+		int16 zone = PChar->getZone();
+
+		const int8* Query = "INSERT INTO server_variables SET name = 'Zone%uDugItems', value = %i ON DUPLICATE KEY UPDATE value = value + %i;";
+		Sql_Query(SqlHandle, Query, zone, value);
+	}
+
+/************************************************************************
+*																		*
+*	Delete the character's Chocobo Dig count							*
+*																		*
+************************************************************************/
+
+	void DeletePlayersDigCounts()
+	{
+		const int8* fmtQuery = "DELETE FROM `char_vars` WHERE 'varname'='dig_count';";
+		Sql_Query(SqlHandle, fmtQuery);
+	}
+
+/************************************************************************
+*																		*
+*	Check if the character is able to Dig								*
+*																		*
+************************************************************************/
+
+	bool CanPlayerDig(CCharEntity* PChar)
+	{
+		uint32 CurrentTime = CVanaTime::getInstance()->getVanaTime();
+		if (GetPlayerDigCount(PChar) < 100)		// Check Player Counter
+		{
+			if (GetZoneItemsDug(PChar) < 20)	// Check Zone Counter
+			{
+				if (GetZoneInTime(PChar) <= (ReturnAreaDigDelay(PChar) + CurrentTime))	// Check Area Dig Delay
+				{
+					if (GetLastDig(PChar) + ReturnDigDelay(PChar) <= CurrentTime)		// Check Character Dig Delay
+					{
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
+
+/************************************************************************
+*																		*
+*	Check if the character's Chocobo is able to Burrow					*
+*																		*
+************************************************************************/
+
+	bool CanPlayerBurrow(CCharEntity* PChar)
+	{
+		// Check for Chocobo Burrow ability here
+		return false;
+	}
+
+/************************************************************************
+*																		*
+*	Check if the character's Chocobo is able to Bore					*
+*																		*
+************************************************************************/
+
+	bool CanPlayerBore(CCharEntity* PChar)
+	{
+		// Check for Chocobo Bore ability here
+		return false;
+	}
+
+/************************************************************************
+*																		*
+*	Check if the character has the Egg Helm equipped					*
+*																		*
+************************************************************************/
+
+	bool EggHelm(CCharEntity* PChar)
+	{
+		CItemArmor* PHead = (CItemArmor*)PChar->getStorage(LOC_INVENTORY)->GetItem(PChar->equip[SLOT_HEAD]);
+		if(PHead->getID() == 16109)
```

Comment:
CanPlayerBore and CanPlayerBurrow were seperated because 'eventually', when chocobo raising is implemented, chocobos are able to learn an ability, either burrow or bore (but not both), and this will check to see if the chocobo knows either one when digging.